src/audio/buffer.c

```diff
@@ -108,9 +108,21 @@ void buffer_free(struct comp_buffer *buffer)
 void comp_update_buffer_produce(struct comp_buffer *buffer, uint32_t bytes)
 {
 	uint32_t flags;
+	int32_t *head;
+	int32_t *tail;
+	uint32_t head_size = bytes;
+	uint32_t tail_size = 0;
 
 	spin_lock_irq(&buffer->lock, flags);
 
+	/* calculate head and tail for dcache circular wrap ops */
+	head = buffer->w_ptr;
+	tail = buffer->addr;
+	if (buffer->w_ptr + bytes >= buffer->end_addr) {
```

Comment:
It doesn't change the behavior, but it's cleaner and spares us additional assignments:
`if (buffer->w_ptr + bytes > buffer->end_addr) {`

Reply:
sorry didn't get this. You mean that works the same without "="?

Reply:
If it's equal, then it just means, that the whole space has been used without a wrap.

Reply:
ok